app/waf/src/models/monitor.model.ts

```diff
@@ -0,0 +1,287 @@
+import {model, property} from '@loopback/repository';
+import {CommonEntity} from './common.model';
+
+@model()
+export class Monitor extends CommonEntity {
+  @property({
+    type: 'string',
+    id: true,
+    required: true,
+  })
+  id: string;
+
+  @property({
+    type: 'number',
+    default: 5,
+  })
+  interval: number;
+
+  @property({
+    type: 'number',
+    default: 16,
+  })
+  timeout: number;
+
+  @property({
+    type: 'string',
+    default: '',
+  })
+  targetAddress: string;
+
+  @property({
+    type: 'number',
+    default: 0,
+  })
+  targetPort: number;
+
+  @property({
+    type: 'string',
+    required: true,
+    default: 'tcp',
+  })
+  monitorType: string;
+
+  // user can specify the below attributes to create a certain type monitor
+
+  @property({
+    type: 'string',
+    default: 'no-error',
+  })
+  acceptRCODE: string;
+
+  @property({
+    type: 'boolean',
+    default: false,
+  })
+  adaptive: boolean;
+
+  @property({
+    type: 'number',
+    default: 500,
+  })
+  adaptiveDivergenceMilliseconds: number;
+
+  @property({
+    type: 'number',
+    default: 100,
+  })
+  adaptiveDivergencePercentage: number;
+
+  @property({
+    type: 'string',
+    default: 'relative',
+  })
+  adaptiveDivergenceType: string;
+
+  @property({
+    type: 'number',
+    default: 1000,
+  })
+  adaptiveLimitMilliseconds: number;
+
+  @property({
+    type: 'number',
+    default: 180,
+  })
+  adaptiveWindow: number;
+
+  @property({
+    type: 'string',
+    default: 'query-type',
+  })
+  answerContains: string;
+
+  @property({
+    type: 'string',
+    default: '',
+  })
+  arguments: string;
+
+  @property({
+    type: 'string',
+  })
+  base: string;
+
+  @property({
+    type: 'boolean',
+    default: true,
+  })
+  chaseReferrals: boolean;
+
+  @property({
+    type: 'string',
+    default: '',
+  })
+  ciphers: string;
+
+  @property({
```

Comment:
Please remove class property. It is only for AS3 JSON, and will be filled during translating.